src/Syncs/IBLT.cpp

```diff
@@ -24,3 +25,81 @@ IBLT::IBLT(long numHashes, long numHashCheck, size_t expectedNumEntries, size_t
     // resize cells to be divisible by number of hash
     hashTable.resize(nEntries + nEntries % numHashes);
 }
+
+bool IBLT::listEntriesandKeys(vector<pair<ZZ, ZZ>> &positive,
+                       vector<pair<ZZ, ZZ>> &negative,
+                       vector<ZZ> &OMSKeys,
+                       vector<ZZ> &SMOKeys) {
+    OMSKeys.clear();
+    SMOKeys.clear();
+    long nErased;
+    do {
+        nErased = 0;
+        for (size_t i = 0; i < hashTable.size(); ++i) {
+            auto& entry = hashTable[i];
+            if (entry.isPure(numHashCheck)) {
+                if (entry.count == 1) {
+                    positive.emplace_back(entry.keySum, entry.valueSum);
+                    OMSKeys.push_back(entry.keySum);  // record key of positive
+                } else {
+                    negative.emplace_back(entry.keySum, entry.valueSum);
+                    SMOKeys.push_back(entry.keySum);  // record key of negative
+                }
+                this->_insert(-entry.count, entry.keySum, entry.valueSum);
+                ++nErased;
+            }
+        }
+    } while (nErased > 0);
```

Comment:
Same questions as before ... and, this looks very similar to the code above - can it be combined into one helper function?